model/task/task.go

```diff
@@ -1372,6 +1385,65 @@ func (t *Task) GetHistoricRuntime() (time.Duration, error) {
 	return time.Duration(runtimes[0].ExpectedDuration), nil
 }
 
+func (t *Task) FetchExpectedDuration() time.Duration {
+	if t.DurationPrediction.TTL == 0 {
+		t.DurationPrediction.TTL = util.JitterInterval(5 * time.Minute)
+	}
+
+	if t.DurationPrediction.Value == 0 && t.ExpectedDuration != 0 {
+		t.DurationPrediction.Value = t.ExpectedDuration
+		t.DurationPrediction.CollectedAt = time.Now().Add(-time.Minute)
+	}
+
+	err := t.DurationPrediction.SetRefresher(func(previous time.Duration) (time.Duration, bool) {
```

Comment:
I don't think that's true, because we're re-querying so the objects in the db are resetting this value... also it'll just be log noise if I'm wrong about this.

Reply:
Doesn't it mean we'll always return `defaultTaskDuration` after the first call?

Reply:
yop